File: apps/backend/src/EnterpriseAgentOs.Domain/Common/ValueObjects/DeviceCodeStatus.cs
namespace EnterpriseAgentOs.Domain.Common.ValueObjects;

public enum DeviceCodeStatus
{
    Pending,
    Authorized,
    Expired,
}

public static class DeviceCodeStatusExtensions
{
    public static string ToStorageString(this DeviceCodeStatus status) => status switch
    {
        DeviceCodeStatus.Pending => "pending",
        DeviceCodeStatus.Authorized => "authorized",
        DeviceCodeStatus.Expired => "expired",
        _ => throw new ArgumentOutOfRangeException(nameof(status)),
    };

    public static DeviceCodeStatus ToDeviceCodeStatus(this string value) => value switch
    {
        "pending" => DeviceCodeStatus.Pending,
        "authorized" => DeviceCodeStatus.Authorized,
        "expired" => DeviceCodeStatus.Expired,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown device code status: {value}"),
    };
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/DeviceCodeStatus.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum DeviceCodeStatus {
    Pending,
    Authorized,
    Expired;

    public String toWire() {
        return switch (this) {
            case Pending -> "pending";
            case Authorized -> "authorized";
            case Expired -> "expired";
        };
    }

    public static DeviceCodeStatus fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "pending" -> Pending;
            case "authorized" -> Authorized;
            case "expired" -> Expired;
            default -> throw new IllegalArgumentException("Unknown device code status: " + value);
        };
    }
}
